ql/src/java/org/apache/hadoop/hive/ql/metadata/PartitionIterable.java

```diff
@@ -99,7 +101,18 @@ private void getNextBatch() {
           batchCounter++;
         }
         try {
-          batchIter = db.getPartitionsByNames(table, nameBatch, getColStats).iterator();
+          if (!isAuthRequired) {
+            batchIter = db.getPartitionsByNames(table, nameBatch, getColStats).iterator();
+          } else {
+            GetPartitionsPsWithAuthRequest req = new GetPartitionsPsWithAuthRequest();
```

Comment:
Can you make this as Global variable GetPartitionsPsWithAuthRequest and reuse

Reply:
Added it in the child method